dap_aria_mapping/pipeline/taxonomy_pruning/preprune_topics.py

```diff
@@ -0,0 +1,174 @@
+import time, pickle, ast, subprocess, re
+from dap_aria_mapping import logger, PROJECT_DIR
+from dap_aria_mapping.getters.taxonomies import (
+    get_cooccurrence_taxonomy,
+)
+from dap_aria_mapping.utils.topic_names import *
+from chatgpt_wrapper import ChatGPT, AsyncChatGPT
+from dap_aria_mapping import chatgpt_args
+
+OUTPUT_DIR = PROJECT_DIR / "outputs" / "interim" / "preprune_topics"
+OUTPUT_DIR.mkdir(parents=True, exist_ok=True)
+
+if __name__ == "__main__":
+    try:
+        logger.info("Loading preprune_topics.pkl")
+        with open(OUTPUT_DIR / "preprune_topics.pkl", "rb") as f:
+            prepruned_entities = pickle.load(f)
+    except FileNotFoundError:
+        logger.info("No preprune_topics.pkl found, creating new list.")
+        prepruned_entities = []
+
+    try:
+        logger.info("Loading entities")
+        with open(OUTPUT_DIR / "entities.pkl", "rb") as f:
+            entities = pickle.load(f)
+        logger.info(f"Loaded entities - Length: {len(entities)}")
+    except FileNotFoundError:
+        logger.info("No entities.pkl found, creating new list.")
+        entities = get_cooccurrence_taxonomy().index.tolist()
+        entities = entities[:35000]
```

Comment:
is this meant to be a hard-coded cut off?

Reply:
No, I had this run on this computers, after I told myself "I'll remember to remove it, it'll be fine". So much for that.